File: routers/classification.go
package routers

import (
	"book-keeping-web/models"
	"book-keeping-web/utils"
	"github.com/gin-gonic/gin"
)

type ClassificationReq struct {
	Name     string `json:"name" binding:"omitempty"`
	LedgerId int    `json:"ledger_id"`
}

// ClassificationCreate 为当前用户创建分类
func ClassificationCreate(c *gin.Context) {
	req := ClassificationReq{}
	if err := c.BindJSON(&req); err != nil {
		utils.FailMessage(c, "parse param error")
		return
	}
	classification := models.Classification{Name: req.Name, UserId: c.GetInt("UserId"), LedgerId: req.LedgerId}
	if !classification.CreateClassification() {
		utils.FailMessage(c, "create classification error")
		return
	}
	utils.SuccessMessage(c, "create classification success")
}

// ClassificationList 查询当前用户的所有分类
func ClassificationList(c *gin.Context) {
	req := ClassificationReq{}
	if err := c.BindJSON(&req); err != nil {
		utils.FailMessage(c, "parse param error")
		return
	}
	var classifications []models.Classification
	classification := models.Classification{UserId: c.GetInt("UserId"), LedgerId: req.LedgerId}
	if !classification.ListClassification(&classifications) {
		utils.FailMessage(c, "list classifications error")
		return
	}
	res := classification.ToClassificationView(&classifications)
	utils.Success(c, gin.H{"classifications": res}, "list classifications success")
}

// ClassificationDelete 删除分类
func ClassificationDelete(c *gin.Context) {
	req := ClassificationReq{}
	if err := c.BindJSON(&req); err != nil {
		utils.FailMessage(c, "parse param error")
		return
	}
	classification := models.Classification{Name: req.Name, UserId: c.GetInt("UserId"), LedgerId: req.LedgerId}
	if !classification.DeleteClassification() {
		utils.FailMessage(c, "delete classification error")
		return
	}
	utils.SuccessMessage(c, "delete classification success")
}

File: frontend/src/api/classification.ts
import axios from './instance/bff_instance'
import { ClassificationId, ClassificationItem, LedgerId } from "./interface";


const classificationApi = {
  listClassification,
  createClassification,
  deleteClassification,
}


/**
 * 查询分类列表
 */
export async function listClassification(ledger_id: LedgerId): Promise<ClassificationItem[]> {
  try {
    const response = await axios.post<{ classifications: ClassificationItem[] }>('/classification/list', {
      ledger_id: ledger_id, // 账本ID
    });
    return response.data.classifications;
  } catch (error) {
    console.error('listClassificationList error', error)
    return [];
  }
}


/**
 * 创建分类
 */
export async function createClassification(ledger_id: LedgerId, name: string): Promise<ClassificationId | null> {
  try {
    const response = await axios.post<ClassificationId>('/classification/create', {
      ledger_id: ledger_id, // 账本ID
      name: name, // 分类名称
    });
    return response.data;
  } catch (error) {
    console.error('createClassification error', error)
    return null;
  }
}

/**
 * 删除分类
 */
export async function deleteClassification(ledger_id: LedgerId, name: string): Promise<boolean> {
  try {
    await axios.post<any>('/classification/delete', {
      ledger_id: ledger_id, // 账本ID
      name: name, // 分类名称
    });
    return true;
  } catch (error) {
    console.error('deleteLedger error', error)
    return false;
  }
}

export default classificationApi